src/selection/join.js

```diff
@@ -3,5 +3,5 @@ export default function(onenter, onupdate, onexit) {
   enter = typeof onenter === "function" ? onenter(enter) : enter.append(onenter + "");
   if (onupdate != null) update = onupdate(update);
   if (onexit == null) exit.remove(); else onexit(exit);
-  return enter && update ? enter.merge(update).order() : update;
+  return enter && update ? enter.selection().merge(update.selection()).order() : update;
```

Comment:
I think we should do this earlier so that *selection*.join is guaranteed to return a selection, like so:

```js
function(onenter, onupdate, onexit) {
  var enter = this.enter(), update = this, exit = this.exit();
  if (typeof onenter === "function") {
    enter =  onenter(enter);
    if (enter) enter = enter.selection();
  } else {
    enter = enter.append(onenter + "");
  }
  if (onupdate != null) {
    update = onupdate(update);
    if (update) update = update.selection();
  }
  if (onexit == null) exit.remove(); else onexit(exit);
  return enter && update ? enter.merge(update).order() : update;
}
```

Reply:
Oh yes good catch! Thank you.